### PR TITLE
feat: simplify withSchema API

### PR DESCRIPTION
**Pull request by ephys**
### Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [x] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

As a follow up to https://github.com/sequelize/sequelize/pull/14337. While we're renaming `Model.schema` to `Model.withSchema`, we might as well simplify its API. I changed it from:

```typescript
MyModel.schema('public');
MyModel.schema('public', /* schema delimiter */ '.');
MyModel.schema('public', { schemaDelimiter: '.' });
```

to

```typescript
MyModel.withSchema('public');
MyModel.withSchema({ schema: 'public', schemaDelimiter: '.' });
```

Not a breaking change because `.schema` keeps its old API and `.withSchema` hasn't been released yet

### CODE REVIEW

File: test/support.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare namespace Chai {
+  interface Assertion {
+    deepEqual(expected: any): Assertion;
```
Comment (WikiRik) [inline test/support.d.ts:3]: Where would we need this for?

**Review by WikiRik — APPROVED** [review]
I have one question, but it's non-blocking

File: test/support.d.ts (hunk 1 above)
Comment (ephys) [reply test/support.d.ts:3]: `support.js` declares a custom chai method:

```
chai.Assertion.addMethod('deepEqual', function (expected, depth = 5) {
  expect(inspect(this._obj, { depth })).to.deep.equal(inspect(expected, { depth }));
});
```

which works like `deep.equal` but includes symbols

This declares it to TypeScript so `expect(y).to.deepEqual(x)` is valid

Comment (WikiRik) [reply test/support.d.ts:3]: Ah, that makes sense

**Pull request merged**